libipl/p10/ipl0.C

```diff
@@ -374,22 +385,106 @@ static int ipl_alignment_check(void)
 	return -1;
 }
 
+static int initialize_and_check_clock_chip(uint8_t& clock_count)
+{
+	struct pdbg_target *clock_target;
+	enum pdbg_target_status status;
+	uint8_t data;
+	int rc = 0;
+
+	// initialize output variable.
+	clock_count = 0;
+	
+	ipl_log(IPL_DEBUG, "Istep: soft reset clock, and verify clock status register\n");
+	pdbg_for_each_class_target("oscrefclk", clock_target) {
+		status = pdbg_target_probe(clock_target);
+		assert(status == PDBG_TARGET_ENABLED);
```

Comment:
avoid assert , this will cerate core dump, should return error back to caller.

Reply:
OSC_REF_CLOCK target maintain HWAS_STATE attribute. , any guard/ deconfigiure use case supportde for this?
then you should eb looking HWAS_STATE based I@C action, Not a good idea to re attempt the failing units agin.

Incase this target support guard , uyou should updating updatehwmdel istepo to set hwas state based on guard state .

Reply:
no, this target doesnt support guard/deconfig